tango/__init__.py

```diff
@@ -12,22 +12,27 @@
     "SqliteDictFormat",
     "Step",
     "step",
+    "StepInfo",
+    "StepInfoSort",
+    "StepState",
     "StepResources",
     "StepCache",
     "StepGraph",
+    "RunSort",
     "Workspace",
 ]
 
-from tango.executor import Executor
-from tango.format import (
+from .executor import Executor
+from .format import (
     DillFormat,
     DillFormatIterator,
     Format,
     JsonFormat,
     JsonFormatIterator,
     SqliteDictFormat,
 )
-from tango.step import Step, StepResources, step
-from tango.step_cache import StepCache
-from tango.step_graph import StepGraph
-from tango.workspace import Workspace
+from .step import Step, StepResources, step
+from .step_cache import StepCache
+from .step_graph import StepGraph
+from .step_info import StepInfo, StepState
+from .workspace import RunSort, StepInfoSort, Workspace
```

Comment:
Aren't relative imports a problem? I remember them breaking in various scenarios.

Reply:
AFAIK the only place where relative imports break is within a `__main__` file, but there are other situations where absolute imports fail yet relative imports do not (https://bugs.python.org/issue25294). For the most part it just comes down to style / personal preference. I prefer using relative imports when importing things from submodules because it's more succinct, while using absolute imports when importing things from parent modules (or submodules of parent modules) because it's more clear. We've adopted that convention in most parts of this library (look at imports in any of our integrations, for example) but not everywhere (`step.py` doesn't use any relative imports).

Reply:
I would love to use relative imports if they didn't break `__main__`. But since they do, why is this safe? Are we saying, the only place they fail is within a `main()` function? So we can use them everywhere else?

I thought it was a problem with `FromParams`, because that relies on class identity, and relative imports make the same class appear under multiple different names.

Reply:
> I would love to use relative imports if they didn't break __main__. But since they do, why is this safe? Are we saying, the only place they fail is within a main() function? So we can use them everywhere else?

It doesn't have to do with the function, it's the module itself. To be more specific, what I meant is that the only place in `tango/` where relative imports won't work is `tango/__main__.py` when `tango` is run directly from the command-line (as opposed to running it as a module with `python -m tango ...`). This is because the module name at runtime of `tango/__main__.py` is `"__main__"`, so it doesn't act like it's inside of the `tango/` module, hence we have to use absolute imports there.

> I thought it was a problem with `FromParams`, because that relies on class identity, and relative imports make the same class appear under multiple different names.

I don't think that's true. It doesn't matter how you import a class, it's still the same class as far as Python is concerned.